Listener.js

```diff
@@ -66,11 +66,30 @@ class Listener {
 
     console.log(`Listening to subscription: ${subscriptionFullName}`);
 
-    const onMessage = (msg) => {
-      onMessageCallback(msg);
-      msg.ack();
+    const onMessageTryCatch = (msg) => {
+      try {
+        onMessageCallback(msg);
+        msg.ack();
+      } catch (err) {
+        console.error(`Error from callback: ${err}\n`);
+        msg.nack();
+        process.exit(1);
+      }
     };
 
+    const onMessageUserHandling = (msg) => {
+      let callback = onMessageCallback(msg);
+      if (!callback.err) {
+          msg.ack();
+        } else {
+        console.error(`Error from callback: ${callback.err}\n`);
+        msg.nack();
+        process.exit(1);
```

Comment:
it's of lesser concern, but process.exit(1) feels a little brutal way of failing. It takes away the ability of the application to cleanly handle the client having a fatal error.